pkg/antctl/raw/check/util.go

```diff
@@ -220,7 +221,7 @@ func Teardown(ctx context.Context, client kubernetes.Interface, clusterName stri
 	fmt.Fprintf(os.Stdout, fmt.Sprintf("[%s] ", clusterName)+"Waiting for Namespace %s to be deleted\n", namespace)
 	err := wait.PollUntilContextTimeout(ctx, 2*time.Second, 1*time.Minute, true, func(ctx context.Context) (bool, error) {
 		_, err := client.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
-		if err != nil {
+		if err != nil && !errors.IsNotFound(err) {
 			return true, nil
 		}
 		return false, nil
```

Comment:
```suggestion
		if err != nil {
			// Return false and the error unless the error is "NotFound"
			if errors.IsNotFound(err) {
				return true, nil // Namespace successfully deleted
			}
			return false, err // Any other error should stop polling
		}
		return false, nil // Namespace still exists, continue polling
```

Reply:
I agree, it looks like the current logic is not correct. The "stop" condition should be `if err != nil && errors.IsNotFound(err)`, which is equivalent to `if errors.IsNotFound(err)`, as we want to stop polling when the resource is no longer found?

Reply:
Thanks to point out, I refined to following codes to tolerate some transmit failures when polling, which means it won't return the error until timeout:

```
err = wait.PollUntilContextTimeout(ctx, 2*time.Second, 1*time.Minute, true, func(ctx context.Context) (bool, error) {
		_, err := client.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
		if err != nil {
			if errors.IsNotFound(err) {
				return true, nil
			}
		}
		return false, nil
	})
```

Reply:
any other error than resource not found should stop polling as well !
if errors.IsNotFound(err) {
				return true, nil // Namespace successfully deleted
			}
			return false, err // Any other error should stop polling
}
		return false, nil

Reply:
Hi @rajnkamr please check my comment, my point is to ignore the error to tolerate some transmit failures in a given timeout instead of returning the error immediately considering this check will run for every PR. It can help to improve the resilience.
@antoninbas please take a look again if you have different thoughts. Thanks.

Reply:
@luolanzone , Ok, for resiliency , in case of other error than errors.IsNotFound(err), let the polling continue.